protocol/rpcprovider/testing.go

```diff
@@ -41,6 +42,43 @@ func validatePortNumber(ipPort string) string {
 	return ipPort
 }
 
+func performCORSCheck(endpoint epochstoragetypes.Endpoint) error {
+	utils.LavaFormatDebug("Checking CORS", utils.Attribute{Key: "endpoint", Value: endpoint})
+	// Construct the URL for the RPC endpoint
+	endpointURL := "http://" + endpoint.IPPORT // Assuming HTTP,
```

Comment:
```suggestion
	endpointURL := "https://" + endpoint.IPPORT // Providers must have HTTPS support,
```

Reply:
also fix the test afterwards as the s will make it fail.

Reply:
fixed here: 67f120c8f0ff8ec4740e5253469ef5cdc2702806